shared/src/main/scala/translit/Russian.scala

```diff
@@ -0,0 +1,115 @@
+package translit
+
+import translit.Helpers._
+
+object Russian extends Language {
+  val uniGrams = Map(
+    'a' -> 'а',
+    'b' -> 'б',
+    'v' -> 'в',
+    'g' -> 'г',
+    'd' -> 'д',
+    'e' -> 'е',
+    'z' -> 'з',
+    'i' -> 'и',
+    'j' -> 'й',
+    'k' -> 'к',
+    'l' -> 'л',
+    'm' -> 'м',
+    'n' -> 'н',
+    'o' -> 'о',
+    'p' -> 'п',
+    'r' -> 'р',
+    's' -> 'с',
+    't' -> 'т',
+    'u' -> 'у',
+    'f' -> 'ф',
+    'x' -> 'х',
+    'h' -> 'х',
+    'c' -> 'ц',
+    'w' -> 'щ',
+    '#' -> 'ъ',
+    'y' -> 'ы'
+  )
+
+  val biGrams = Map(
+    "jo" -> 'ё',
+    "yo" -> 'ё',
+    "zh" -> 'ж',
+    "ch" -> 'ч',
+    "sh" -> 'ш',
+    "ye" -> 'э',
+    "yu" -> 'ю',
+    "ju" -> 'ю',
+    "ya" -> 'я',
+    "ja" -> 'я'
+  )
+
+  val triGrams = Map(
+    "shh" -> 'щ'
+  )
+
+  // tried to use prefix rules but there are many exceptions in Russian language
+  // Ex.: фольклор, пальцем
+  val apostropheSuffix = Set(
```

Comment:
As far as I understand you want to detect soft vs. hard sign. The most reliable way is to derive those rules from a larger text corpus such as Wikipedia. For now, we can remove those two sets and use designated characters for soft/hard signs.